tests/integration/replication.tcl

```diff
@@ -546,44 +535,82 @@ test {Diskless load swapdb doesn't cause LOADING response when master replid is
+                }
             }
 
-            # Check that classic loading flag is NOT set to 1
-            assert_equal [s -1 loading] 0
-            
-            # Ensure we still see old values while async_loading is in progress and also not LOADING status
-            assert_equal [$replica get mykey] "myvalue"
+            switch $testType {
+                "Aborted" {
+                    # Make sure that next sync will not start immediately so that we can catch the replica in between syncs
+                    $master config set repl-diskless-sync-delay 5
 
-            # Make sure we're still async_loading to validate previous assertion
-            assert_equal [s -1 async_loading] 1
+                    # Kill the replica connection on the master
+                    set killed [$master client kill type replica]
 
-            # Wait for loading to stop
-            wait_for_condition 50 100 {
-                [s -1 async_loading] eq 0
-            } else {
-                fail "Loading didn't stop"
-            }
+                    # Wait for loading to stop (fail)
+                    wait_for_condition 100 100 {
+                        [s -1 async_loading] eq 0
+                    } else {
+                        fail "Replica didn't disconnect"
+                    }
+                    
+                    test {Diskless load swapdb (async_loading): old database is exposed after async replication fails} {
+                        # Ensure we see old values from replica
+                        assert_equal [$replica get mykey] "myvalue"
 
-            # Enable fast shutdown
-            $master config set rdb-key-save-delay 0
+                        # Make sure amount of replica keys didn't change
+                        assert_equal [$replica dbsize] 2001
+                    }
+                }
+                "Delayed" {
+                    test {Diskless load swapdb (async_loading): replica enter async_loading} {
+                        assert_equal [s -1 async_loading] 1
+                        assert_equal [s -1 loading] 0
+                    }
+                    
+                    test {Diskless load swapdb (async_loading): old database is exposed while async replication is in progress} {
+                        # Ensure we still see old values while async_loading is in progress and also not LOADING status
+                        assert_equal [$replica get mykey] "myvalue"
+
+                        # Make sure we're still async_loading to validate previous assertion
+                        assert_equal [s -1 async_loading] 1
 
-            # Ensure we don't see anymore the key that was stored only to replica and also that we don't get LOADING status
-            assert_equal [$replica GET mykey] ""
+                        # Make sure amount of replica keys didn't change
+                        assert_equal [$replica dbsize] 2001
+                    }
```

Comment:
same here..
i.e. on one hand, we wanna test 3 things:
1. state during loading
2. after it is aborted
3. and when it succeeds

so it is nice to see 3 tests with 3 titles.
but on the other hand, we can combine the the first two and save some time.